src/main/java/pl/psnc/dei/util/EuropeanaRecordTransferValidationUtil.java

```diff
@@ -48,16 +48,19 @@ public static String getMimeType(JsonObject record) {
 	 *
 	 * @param record   record json-ld object
 	 * @param mimeType record's mimeType
-	 * @return {@link TransferPossibility}
+	 * @return {@link IiifAvailability}
 	 */
-	public static TransferPossibility checkIfTransferPossible(JsonObject record, String mimeType) {
+	public static IiifAvailability checkIfIiifAvailable(JsonObject record, String mimeType) {
+		if (record == null) {
+			return IiifAvailability.DATA_UNAVAILABLE;
+		}
 		if (checkIfIiif(record)) {
-			return TransferPossibility.POSSIBLE;
+			return IiifAvailability.AVAILABLE;
 		}
 		if (Arrays.asList(ALLOWED_TYPES).contains(mimeType)) {
```

Comment:
The method for checking if IIIF is available is executed for each search result. And for each execution ALLOWED_TYPES array is converted to a List. Why don't you simply make it a list to avoid this conversion?